mcc/odin/odin/odin.go

```diff
@@ -124,15 +125,30 @@ func CreateDBInstance(
 	params CreateDBParams,
 	svc rdsiface.RDSAPI,
 ) (result string, err error) {
-	rdsParams := params.GetCreateDBInstanceInput(
-		instanceName,
-		svc,
-	)
-	res, err := svc.CreateDBInstance(rdsParams)
-	if err != nil {
-		return
+	var instance rds.DBInstance
+	if params.Restore {
+		rdsParams := params.GetRestoreDBInstanceFromDBSnapshotInput(
+			instanceName,
+			svc,
+		)
+		res, err2 := svc.RestoreDBInstanceFromDBSnapshot(rdsParams)
+		if err2 != nil {
+			err = err2
```

Comment:
What about declaring res beforehand to avoid the `:=`?
```
var res rdsiface.RDSAPI.RestoreDBInstanceFromDBSnapshotOutput
res, err = svc.RestoreDBInstanceFromDBSnapshot(rdsParams)
if err != nil {
        return
}
```